### DISCUSSION TITLE
Close trade after specific time

### DISCUSSION THREAD

**Discussion by McHill007** [kernc/backtesting.py#505] [Q&A] [answered]
Hi, sorry für the question but I'm not able to find a nice solution. 

I would like to hold every trade for 9 months ~ 180 candles, with a stop loss of 20%.

Could somebody give me a hint how to solve this in a nice way.

Thanks
Eric

**Comment by McHill007** [comment, marked as answer]
I managed to implement the hold time. 

```
        for trade in self.trades:
            d0 = trade.entry_time.to_pydatetime().date()
            d1 = self.data.index[-1].date()
            delta = d1 - d0
            if (  delta.days )  >= 180:
                trade.close()    
```

**Reply by kernc** [reply]
You could use `pd.Timestamp` (and `pd.Timedelta`) directly:
```py
    if (self.data.index[-1] - trade.entry_time).days > 180:
        ...
```
For a stop loss, pass:
```py
    self.buy(..., sl=0.8 * cur_price)
```

**Reply by McHill007** [reply]
Ok, that's nicer ... thanks for support :)

How could I trail the 20%?

**Reply by McHill007** [reply]
seems like this works out
`TSL = max(trade.sl,slprice)`

**Reply by kernc** [reply]
Exactly. It's what [`TrailingStrategy`](https://kernc.github.io/backtesting.py/doc/backtesting/lib.html#backtesting.lib.TrailingStrategy) does:
https://github.com/kernc/backtesting.py/blob/ae11c904f46edacdf0eb602ad1f1ecd765195fcc/backtesting/lib.py#L476-L486

**Comment by acierum** [comment]
thanks! will give that a try
